dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/CoreSignInJourneyWithTrnLookup.cs

```diff
@@ -1,47 +1,159 @@
 using Microsoft.AspNetCore.Mvc;
-using TeacherIdentity.AuthServer.Models;
+using Microsoft.EntityFrameworkCore;
+using TeacherIdentity.AuthServer.Oidc;
 
 namespace TeacherIdentity.AuthServer.Journeys;
 
-public class CoreSignInJourneyWithTrnLookup : SignInJourney
+public class CoreSignInJourneyWithTrnLookup : CoreSignInJourney
 {
-    public CoreSignInJourneyWithTrnLookup(HttpContext httpContext, IdentityLinkGenerator linkGenerator)
-        : base(httpContext, linkGenerator)
+    private readonly TrnLookupHelper _trnLookupHelper;
+
+    public CoreSignInJourneyWithTrnLookup(
+        HttpContext httpContext,
+        IdentityLinkGenerator linkGenerator,
+        CreateUserHelper createUserHelper, TrnLookupHelper trnLookupHelper)
+        : base(httpContext, linkGenerator, createUserHelper)
     {
+        _trnLookupHelper = trnLookupHelper;
     }
 
-    public override bool CanAccessStep(string step)
+    public override async Task<IActionResult> TryCreateUser(string currentStep)
     {
-        throw new NotImplementedException();
+        try
+        {
+            var user = await CreateUserHelper.CreateUserWithTrn(AuthenticationState);
+
+            AuthenticationState.OnTrnLookupCompletedAndUserRegistered(user);
+            await AuthenticationState.SignIn(HttpContext);
+
+            if ((!AuthenticationState.TryGetOAuthState(out var oAuthState) || !oAuthState.HasScope(CustomScopes.Trn)) &&
+                AuthenticationState.TrnLookupStatus == TrnLookupStatus.Pending)
+            {
+                await CreateUserHelper.CreateTrnResolutionZendeskTicket(AuthenticationState);
+            }
+        }
+        catch (DbUpdateException dex) when (dex.IsUniqueIndexViolation("ix_users_trn"))
+        {
+            // We don't currently handle duplicate TRNs in Core Sign In Journey
+            return new BadRequestResult();
+        }
+
+        return new RedirectResult(GetNextStepUrl(currentStep));
```

Comment:
Was trying to keep redirection logic out of CreateUserHelper without much success